install_config/aggregate_logging.adoc

```diff
@@ -12,628 +12,482 @@ toc::[]
-          type file
-          path /var/log/td-agent/forward-failed
-        </secondary>
-      </store>
+ifdef::openshift-enterprise[]
+The template is available on install, and can be created with the following
+command:
 
-      <store>
-        type file
-        path /var/log/td-agent/containers.log
-        time_slice_format %Y%m%d
-        time_slice_wait 10m
-        time_format %Y%m%dT%H%M%S%z
-        compress gzip
-        utc
-      </store>
-    </match>
 ----
-<1> The name for the master that will be used during logging.
-<2> The IP or a DNS resolvable name used to access the master.
-====
+$ oc create -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/infrastructure-templates/enterprise/logging-deployer.yaml
+----
+endif::openshift-enterprise[]
 
-. Enable *fluentd*:
+ifdef::openshift-origin[]
+The template can be created with the following command:
 
-====
 ----
-# systemctl enable td-agent
-# systemctl start td-agent
+$ oc create -f https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_examples/files/examples/infrastructure-templates/origin/logging-deployer.yaml
 ----
-====
+endif::openshift-origin[]
+
+
 
-[TIP]
+[IMPORTANT]
 ====
-Any errors are logged in the *_/var/log/td-agent/td-agent.log_* file.
+The OpenShift Authentication Proxy (*openshift-auth-proxy*) depends on a *node*
+base image, which is intended to be the DockerHub Node.js base image. Defining
+standard templates also depends on a Node.js image called *node*, which will be
+used instead of the intended image, causing the build to fail. To resolve this,
+delete the older *node* image:
+
+----
+$ oc delete is/node -n openshift
+----
 ====
 
-[[optional-method-to-verify-working-nodes]]
+After defining any builds, they should restart. If not, you can restart each
+component:
 
-=== Optional Method to Verify Working Nodes
+----
+$ oc start-build <component>
+----
 
-You can optionally set up the master to be the aggregator to test and verify
-that the nodes are working properly.
+Once the builds have successfully completed, the image streams will be populated
+and ready for deployment. However, you must first specify the `*IDEX_PREFIX*`
+parameter with the registry location.
 
-. Install *fluentd* (*td-agent*) on the master:
+. Run the deployer, specifying the host on which kibana will run, and the master
+URL where browsers will be directed to when authenticating to OpenShift. Be sure
+to replace the `*KIBANA_HOSTNAME*` and `*PUBLIC_MASTER_URL*` to values relevant
+to your deployment:
 +
 ====
 ----
-# export RPM=td-agent-2.2.0-0.x86_64.rpm
-# curl https://packages.treasuredata.com/2/redhat/7/x86_64/$RPM \
-    -o /tmp/$RPM
-# yum localinstall $RPM
-# mkdir -p /etc/td-agent/config.d
-# chown td-agent:td-agent /etc/td-agent/config.d
+$ oc process logging-deployer-template -n openshift \
+           -v KIBANA_HOSTNAME=kibana.example.com,PUBLIC_MASTER_URL=https://localhost:8443 \
+           | oc create -f -
```

Comment:
It looks like `ES_CLUSTER_SIZE` is also required:

```
# oc process logging-deployer-template             -v KIBANA_HOSTNAME=adellape-111115-quick.os1.phx2.redhat.com,PUBLIC_MASTER_URL=https://adellape-111115-quick.os1.phx2.redhat.com:8443            | oc create -f -
error processing the template "logging-deployer-template": template "logging-deployer-template" is invalid: parameters: invalid value '{Name:ES_CLUSTER_SIZE DisplayName: Description:How many instances of ElasticSearch to deploy. Value: Generate: From: Required:true}', Details: template.parameters[8]: parameter ES_CLUSTER_SIZE is required and must be specified
error: no objects passed to create
```

I set it to 1 and it created fine.